src/main/webapp/js/instructorHome.es6

```diff
@@ -107,3 +120,42 @@ function instructorHomeDateComparator(x, y) {
     }
     return x0 < y0 ? -1 : 0;
 }
+
+function bindCoursePanels() {
+    const panels = $('div.panel');
+    let numPanels = 0;
+    for (let i = 0; i < panels.length; i += 1) {
+        const heading = $(panels[i]).children('.panel-heading');
+        const bodyCollapse = $(panels[i]).children('.panel-collapse');
+        if (heading.length !== 0 && bodyCollapse.length !== 0) {
+            $(heading[0]).attr('data-target', `#panelBodyCollapse-${numPanels}`);
+            $(heading[0]).attr('id', `panelHeading-${numPanels}`);
+            $(heading[0]).css('cursor', 'pointer');
+            $(bodyCollapse[0]).attr('id', `panelBodyCollapse-${numPanels}`);
+        }
+        numPanels += 1;
+    }
+}
+
+/**
+ * Changes the state of the course panel (collapsed/expanded).
+ */
+function toggleSingleCourseCollapse(e) {
+    if ($(e.target).is('a') || $(e.target).is('input') || $(e.target).hasClass('dropdown-toggle')) {
+        return;
+    }
+    const glyphIcon = $(this).find('.glyphicon');
+    const className = $(glyphIcon[0]).attr('class');
+    const dropdowns = $(this).find('.dropdown');
+    if (className.indexOf('glyphicon-chevron-up') === -1) {
```

Comment:
I referenced the existing functionality of expanding/collapsing functionality and every where icon state is checked through its class. May be we can use hidden variable for each panel for the icon state.